### PR TITLE
Add logDirectory option to copy logs to WACZ

### LINKED ISSUE #118: Add option to copy log files from specified directory into WACZ
This will aid with [parity with py-wacz](https://github.com/webrecorder/py-wacz?tab=readme-ov-file#-l---log-directory) as well as making the switch to using js-wacz in Browsertrix Crawler.

Related to https://github.com/webrecorder/browsertrix-crawler/issues/484

PR forthcoming in the next day or so :)

### PR DESCRIPTION
**Pull request by tw4l**
Fixes #118 

Thanks for taking a look, this is one of the last things needed for Browsertrix Crawler!

### CODE REVIEW

**Review by matteocargnelutti — CHANGES_REQUESTED** [review]
@tw4l This looks great as usual 🤝 ! 

One question and one request: 
- Is there a common pattern for these log files (extension, content, etc ...)? I would be interested in making that feature less permissive, so that we don't have a "add whatever to WACZ" flag 😅 . 
- Would you mind adding docs for this new option to `types.js` so the IDE / JSDoc knows about this new option? 

Thank you!

**Comment by tw4l** [comment]
>     * Is there a common pattern for these log files (extension, content, etc ...)? I would be interested in making that feature less permissive, so that we don't have a "add whatever to WACZ" flag 😅 .

The log files from Browsertrix Crawler are JSON-L files with `.log` as the extension - happy to add extra validation there. Perhaps we can limit to `log` or `txt` extensions for now? I think if there was an issue with the correctness of the log files (not that I've seen that happen, but you never know) it'd still be better to include a file with a formatting issue rather than no logs.

>     * Would you mind adding docs for this new option to `types.js` so the IDE / JSDoc knows about this new option?

Happy to!

**Comment by matteocargnelutti** [comment]
@tw4l 
> The log files from Browsertrix Crawler are JSON-L files with .log as the extension - happy to add extra validation there. Perhaps we can limit to log or txt extensions for now? I think if there was an issue with the correctness of the log files (not that I've seen that happen, but you never know) it'd still be better to include a file with a formatting issue rather than no logs.

That makes sense. Let's start by only accepting `.log` and `.txt` files. Thank you :)

File: index.js [hunk 1]
```diff
@@ -185,6 +185,12 @@ export class WACZ {
    */
   pagesDir = null
 
+  /**
+   * Path to directory of log files to copy into WACZ.
+   * @type {?string}
+   */
+  logDirectory = null
```
[the remark this answers is not in the document]
Comment (tw4l) [reply index.js]: Done! I kept the option for class initialization as `logDirectory` and CLI flag as log-directory for compatibility with py-wacz but internally it now uses `this.logDir`

[the remark this answers is not in the document]
Comment (tw4l) [reply index.js]: Feel free to change the former though, I don't have a strong opinion, as long as the CLI flag is the same I think we're good :)

**Comment by tw4l** [comment]
Rebased and updated, thanks!

**Comment by matteocargnelutti** [comment]
Hey there @tw4l! This is great, thanks! 
I pushed a commit for suggest an updated naming convention for all options that expect / accept a path to dir - what do you think? 

Cheers,

**Comment by tw4l** [comment]
> Hey there @tw4l! This is great, thanks! I pushed a commit to suggest an updated naming convention for all options that expect / accept a path to dir - what do you think?

Hi @matteocargnelutti , makes sense to me! The consistency is nice :) Thanks for this!

**Review by matteocargnelutti — APPROVED** [review]

**Pull request merged**